Cabal/src/Distribution/Simple/GHC.hs

```diff
@@ -214,6 +215,29 @@ configure verbosity hcPath hcPkgPath conf0 = do
       progdb4 = Internal.configureToolchain implInfo ghcProg ghcInfoMap progdb3
   return (comp, compPlatform, progdb4)
 
+type GhcInfo = Map String String
```

Comment:
Can this get a haddockumentation?  
- What is the purpose?
- What are the keys?
- What are the values?
- Example?

Reply:
It's the output of `ghc --info`